tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -112,13 +132,116 @@ public static Security Instance
 
         internal SecuritySettings Settings => _settings;
 
-        internal string DdlibWafVersion => _waf?.Version;
+        internal string? DdlibWafVersion => _waf?.Version;
+
+        internal void SubscribeToChanges(params string[] productNames)
+        {
+            void RcmSubscription(RemoteConfigurationManager remoteConfigurationManager)
+            {
+                _remoteConfigurationManager ??= remoteConfigurationManager;
+                if (_rcmSubscription is null)
+                {
+                    _rcmSubscription = remoteConfigurationManager.SubscribeToChanges(
+                        UpdateFromRcm,
+                        productNames);
+                }
+                else
+                {
+                    _rcmSubscription.SubscribeProducts(productNames);
+                }
+            }
+
+            if (_remoteConfigurationManager is null)
+            {
+                RemoteConfigurationManager.CallbackWithInitializedInstance(RcmSubscription);
+            }
+            else
+            {
+                RcmSubscription(_remoteConfigurationManager);
+            }
+        }
 
-        internal BlockingAction GetBlockingAction(string id, string[] requestAcceptHeaders)
+        internal List<ApplyDetails> UpdateFromRcm(Dictionary<string, List<RemoteConfiguration>> configsByProduct, Dictionary<string, List<RemoteConfigurationPath>>? removedConfigs)
```

Comment:
Internal note: Can this be happenning on multiple threads? I don't think so but need to check